lv_draw/lv_draw_rbasic.c

```diff
@@ -302,16 +302,19 @@ void lv_rmap(const lv_area_t * cords_p, const lv_area_t * mask_p,
     } else {
         lv_color_t chroma_key_color = LV_COLOR_TRANSP;
         lv_coord_t col;
+        lv_color_t img_buffer[masked_a.x2 - masked_a.x1];
```

Comment:
A few comments here:
1) Where possible, please avoid use of dynamically sized arrays. This could easily be a _huge_ array and lead to us blowing the stack.
2) Please reuse the temporary pixel buffer used in the character draw acceleration. That buffer size is configured at compile time, to allow us to control stack usage.
3) Since the buffer is likely not going to be the exact size of your image, you will need to implement a sort of "chunking". E.g. if the buffer is only 1KB, but your image is 10KB, you need to do 10 chunked image draws 1KB at a time.

Reply:
Take a look at the temporary buffer in `lv_rletter()`:
https://github.com/GridPlus/lvgl/blob/3d93bcbe675db87e2b8e134393d8cbf8cd66eeff/lv_draw/lv_draw_rbasic.c#L124

This buffer is controlled by `#define`s in `lv_conf.h`. This lets us optionally disable the static buffer and/or adjust its allocated size.

For this feature to be complete, you should reuse this buffer, so that we do not have multiple bespoke runtime buffers. Perhaps move the buffer out of `lv_rletter()` and make it a shared static buffer that can be used in both places (and other places in the future).

Reply:
Okay. Made the changes you outline above. Called it `pixel_buffer[]` .Turns out lv_disp_map() only draws within the rect you specify, so no need to worly about chunking.